src/gt4py/next/embedded/nd_array_field.py

```diff
@@ -254,3 +371,87 @@ class JaxArrayField(_BaseNdArrayField):
         array_ns: ClassVar[ModuleType] = jnp
 
     common.field.register(jnp.ndarray, JaxArrayField.from_array)
+
+
+def _find_index_of_dim(dim: Dimension, domain_slice: DomainSlice) -> Optional[int]:
+    for i, (d, _) in enumerate(domain_slice):
+        if dim == d:
+            return i
+    return None
+
+
+def broadcast(field: common.Field, new_dimensions: tuple[common.Dimension, ...]):
+    domain_slice = []
+    new_domain_dims = []
+    new_domain_ranges = []
+    for dim in new_dimensions:
+        if (pos := _find_index_of_dim(dim, field.domain)) is not None:
+            domain_slice.append(slice(None))
+            new_domain_dims.append(dim)
+            new_domain_ranges.append(field.domain[pos][1])
+        else:
+            domain_slice.append(np.newaxis)
+            new_domain_dims.append(dim)
+            new_domain_ranges.append(
+                UnitRange(common.Infinity.negative(), common.Infinity.positive())
+            )
+    return common.field(
+        field.ndarray[tuple(domain_slice)],
+        domain=Domain(tuple(new_domain_dims), tuple(new_domain_ranges)),
+    )
+
```

Comment:
```suggestion
_BaseNdArrayField.register_builtin_func(fbuiltins.broadcast, broadcast)
```
this registers the broadcast to be used as a field function. please change the broadcast tests to use the `fbuiltins.broadcast`.
Also test broadcasting a value to a field, this might require modifications to the broadcast function.

Reply:
broadcasting scalars actually doesn't reach this function (and we wouldn't know which ndarray to construct)